js/boaexchange.js

```diff
@@ -0,0 +1,1163 @@
+                    'maker': 0.15,
+                    'taker': 0.15,
+                },
+                'funding': {
+                    'tierBased': false,
+                    'percentage': false,
+                    'withdraw': {
+                    },
+                    'deposit': {
+                    },
+                },
+            },
+            'exceptions': {
+                'Invalid X-BOA-ENCRYPTED': AuthenticationError,
+                'Invalid X-BOA-NONCE': InvalidNonce,
+                'Address is Pending': AddressPending,
+                'Insufficient Funds': InsufficientFunds,
+                'Order Not Found': OrderNotFound,
+                'Too Many Requests': DDoSProtection,
+            },
+            'options': {
+                'hasEmptyMarkets': true,
+                // price precision by quote currency code
+                'pricePrecisionByCode': {
+                    'USD': 3,
+                },
+                'symbolSeparator': '_',
+                'tag': {
+                },
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    costToPrecision (symbol, cost) {
+        return this.decimalToPrecision (cost, TRUNCATE, this.markets[symbol]['precision']['price'], DECIMAL_PLACES);
+    }
+
+    feeToPrecision (symbol, fee) {
+        return this.decimalToPrecision (fee, TRUNCATE, this.markets[symbol]['precision']['price'], DECIMAL_PLACES);
+    }
+
+    nonce () {
+        return this.seconds ();
+    }
+
+    async createDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'label': currency['id'],
+        };
+        const response = await this.v1GetAddressesLabel (this.extend (request, params));
+        if (response['data'].length > 0) {
+            return this.parseDepositAddress (response['data'][0]);
+        }
+        return [];
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        if (type !== 'limit')
+            throw new ExchangeError (this.id + ' allows limit orders only');
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let order = {
+            'market': market['id'],
+            'amount': this.amountToPrecision (symbol, amount),
+            'price': this.priceToPrecision (symbol, price),
+            'side': side,
+        };
+        let response = await this.v1GetOrder (this.extend (order, params));
+        return this.extend (this.parseOrder (response['data'], market), {
+            'status': 'open',
+            'price': order['price'],
+            'symbol': symbol,
+            'amount': order['amount'],
+            'side': side,
+            'type': type,
+            'id': response['data']['id'],
+        });
+    }
+
+    async createLimitOrder (symbol, type, side, amount, price = undefined, params = {}) {
```

Comment:
This override looks redundant, no?

Reply:
At the moment, i believe the exchange is working on supporting market orders which is why i added it since i'd need to override it once theyve enabled it. but i can remove it until then

Reply:
I'd rather just indicate `createOrder and/or createLimitOrder and/or createMarketOrder = true/false` in describe() with the actual values, until this is added on the exchange side. Does that sound ok?